### ISSUE TITLE
Followup from #10063 - timeseries operations on multi-indexed dataframes still broken

### ISSUE THREAD

**Issue by neiseman** [pydata/pandas#10128]
#10063 doesn't appear to have been completely fixed. While the exact example I gave now works (the resample), what if instead of a resample I had wanted to do a tshift? I would have wanted to be able to do df.groupby(level='symbol').tshift(1, 'D'), which still returns an empty DataFrame. Here, there's no aggregating in the time dimension so the TimeGrouper syntax above wouldn't make sense.

I also don't understand why the `df.groupby([pd.Grouper(level='symbol'), pd.Grouper(level='dt', freq='M')]).mean()` syntax is necessary instead of simply being able to do `df.groupby(level='symbol').resample('M', how='mean')`? The latter seems like much better design.

**Comment by jreback** [comment]
@neiseman doing a chained groupby operation is in general not a good idea. It is completely non-performant compared to a multi-groupby operation. 

You can if you want do:

``df.groupby(...).apply(lambda x: x.resample(...))`` to achieve this.

There is really no way to have a chained syntax be lazy (well you can actually do this via [blaze](http://blaze.pydata.org/en/latest/)) to some extent.

**Comment by shoyer** [comment]
In principle, I think `df.groupby(level='symbol').resample('M', how='mean')` could consolidate the two groupby operations under the hood. This might require some refactoring of the logic in `pandas.core.groupby`, though.

**Comment by neiseman** [comment]
I agree that it would be nice if the groupby operations could be transparently consolidated. Failing that, trying something like `df.groupby(level='symbol').resample('M', how='mean')` or `df.groupby(level='symbol').tshift(1, 'D')` should raise an error that you can't chain groupbys--definitely not return an empty DataFrame.

**Comment by shoyer** [comment]
@neiseman If you're like to put together a PR to add those error messages, that would be appreciated :)

**Comment by mroeschke** [comment]
It looks like this properly raises an error now:

```
NotImplementedError: Not supported for type MultiIndex
```

We can change this issue to support `groupby.resample` for MultiIndex DataFrames
